app/src/main/java/com/kamron/pogoiv/pokeflycomponents/IVPreviewPrinter.java

```diff
@@ -73,7 +73,8 @@ public void run() {
             boolean succeeded = runQuickScan();
             Pokefly pokefly = pokeflyRef.get();
             if (!succeeded && pokefly != null) {
-                Toast.makeText(pokefly, R.string.touch_screen_quick_iv_scan, Toast.LENGTH_SHORT).show();
+                ivButtonRef.get().showError();
```

Comment:
To avoid a null pointer exception when the IV button gets garbage collected, ivButtonRef.get() return value should be checked, also Pokefly instance isn't needed anymore since we disabled the Toast and thus there's no need for the Context instance:

```
boolean succeeded = runQuickScan();
IvButton ivButton = ivButtonRef.get();
if (!succeeded && ivButton != null) {
    ivButton.showError();
}
```